model/src/ratio_as_decimal.rs

```diff
@@ -0,0 +1,139 @@
+use bigdecimal::BigDecimal;
+use num::bigint::Sign as Sign04;
+use num::BigRational;
+use num_bigint::{BigInt, Sign as Sign03};
+use serde::{de, Deserializer, Serializer};
+use serde_with::{DeserializeAs, SerializeAs};
+use std::fmt;
+use std::str::FromStr;
+
+pub struct DecimalBigRational;
+
+impl<'de> DeserializeAs<'de, BigRational> for DecimalBigRational {
+    fn deserialize_as<D>(deserializer: D) -> Result<BigRational, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserialize(deserializer)
+    }
+}
+
+impl<'de> SerializeAs<BigRational> for DecimalBigRational {
+    fn serialize_as<S>(source: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serialize(source, serializer)
+    }
+}
+
+pub fn serialize<S>(value: &BigRational, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    let top_bytes = value.numer().to_bytes_le();
+    let top = BigInt::from_bytes_le(sign_04_to_03(top_bytes.0), &top_bytes.1);
+    let bottom_bytes = value.denom().to_bytes_le();
+    let bottom = BigInt::from_bytes_le(sign_04_to_03(bottom_bytes.0), &bottom_bytes.1);
+    let decimal = BigDecimal::from(top) / BigDecimal::from(bottom);
+    serializer.serialize_str(&decimal.to_string())
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<BigRational, D::Error>
```

Comment:
I'm not sure this implementation is correct, I would expect:
```
assert!(ratio_as_decimal::deserialize(json!("-1/1")).is_err());
```

Because `-1/1` is not a decimal string.

Reply:
oh yea... weird...